src/core/gltf2importer/gltf2parser.cpp

```diff
@@ -485,6 +519,85 @@ const GLTF2Context *GLTF2Parser::context() const
+{
+    QByteArray tangentBuffer;
+    const int bufferIndex = rootObject.value(KEY_BUFFERS).toArray().size();
+    for (int i = 0; i < m_context->meshesCount(); ++i) {
+        const Mesh mesh = m_context->mesh(i);
+
+        for (int nPriv = 0; nPriv < mesh.meshPrimitives.size(); ++nPriv) {
+            Qt3DRender::QGeometryRenderer *primitive = mesh.meshPrimitives[nPriv].primitiveRenderer;
+
+            Qt3DRender::QAttribute *newTangentAttr = GLTF2Import::MeshParserUtils::createNewTangentAttributeIfNeeded(primitive);
+            if (!newTangentAttr)
+                continue;
+
+            const QByteArray buffer = newTangentAttr->buffer()->data();
+
+            // add bufferView
+            QJsonObject jsonBufferView;
+            jsonBufferView[KEY_BUFFER] = bufferIndex;
+            jsonBufferView[KEY_BYTELENGTH] = buffer.size();
+            jsonBufferView[KEY_BYTEOFFSET] = tangentBuffer.size();
+            jsonBufferView[KEY_BUFFERVIEWTARGET] = GL_ARRAY_BUFFER;
+            const int bufferViewIndex = addToJsonArray(rootObject, KEY_BUFFERVIEWS, jsonBufferView);
+
+            tangentBuffer.append(buffer);
+
+            // add accessor
+            QJsonObject jsonAccessor;
+            jsonAccessor[KEY_BUFFERVIEW] = bufferViewIndex;
+            jsonAccessor[KEY_COMPONENTTYPE] = GL_FLOAT;
+            jsonAccessor[KEY_COUNT] = static_cast<int>(newTangentAttr->count());
+            jsonAccessor[KEY_TYPE] = QLatin1Literal("VEC4");
+            jsonAccessor[KEY_MIN] = QJsonArray{0.0, 0.0, 0.0, 0.0};
+            jsonAccessor[KEY_MAX] = QJsonArray{0.0, 0.0, 0.0, 0.0};
+            const int accessorIndex = addToJsonArray(rootObject, KEY_ACCESSORS, jsonAccessor);
+
+            // modify mesh object and add attribute/accessor
+            // 1st search for mesh JSON object
+            QJsonArray jsonMeshes = rootObject.value(KEY_MESHES).toArray();
+            for (auto it = jsonMeshes.begin(); it != jsonMeshes.end(); ++it) {
+                QJsonObject jsonMesh = it->toObject();
+                if (jsonMesh.value(KEY_NAME) != mesh.name)
+                    continue;
+
+                // replace matching primitive
+                QJsonArray jsonPrimitives = jsonMesh.value(KEY_PRIMITIVES).toArray();
+                QJsonObject jsonPrimitive = jsonPrimitives.at(nPriv).toObject();
+                QJsonObject jsonPrimAttrs = jsonPrimitive.value(KEY_ATTRIBUTES).toObject();
+                jsonPrimAttrs[ATTR_TANGENT] = accessorIndex;
+                jsonPrimitive[KEY_ATTRIBUTES] = jsonPrimAttrs;
+                jsonPrimitives[nPriv] = jsonPrimitive;
+                jsonMesh[KEY_PRIMITIVES] = jsonPrimitives;
+
+                *it = jsonMesh;
+            }
+            rootObject[KEY_MESHES] = jsonMeshes;
+        }
+    }
+
+    // add tangent buffer file
+    if (!tangentBuffer.isEmpty()) {
+        // create file
+        const QLatin1Literal bufferFileName("kuesaTangentBuffers.bin");
```

Comment:
this function could be useful : https://github.com/KDAB/kuesa/blob/dev/src/core/gltf2exporter/gltf2exporter_p.cpp#L55 (else potentially if loading / saving an asset stuff may be overwritten).